lib/spack/spack/schema/container.py

```diff
@@ -16,35 +53,22 @@
         },
         # Describes the base image to start from and the version
         # of Spack to be used
-        'base': {
-            'type': 'object',
-            'additionalProperties': False,
-            'properties': {
-                'image': {
-                    'type': 'string',
-                    'enum': ['ubuntu:18.04',
-                             'ubuntu:16.04',
-                             'centos:7',
-                             'centos:6']
-                },
-                'spack': {
-                    'type': 'string',
-                    'enum': ['develop', '0.14', '0.14.0']
-                }
-            },
-            'required': ['image', 'spack']
-        },
+        'images': {'anyOf': [_stages_from_dockerhub, _custom_stages]},
         # Whether or not to strip installed binaries
         'strip': {
             'type': 'boolean',
             'default': True
         },
         # Additional system packages that are needed at runtime
         'os_packages': {
-            'type': 'array',
-            'items': {
-                'type': 'string'
-            }
+            'type': 'object',
+            'properties': {
+                'command': {'type': 'string', 'enum': ['apt', 'yum']},
+                'update': {'type': 'boolean'},
+                'build': _list_of_packages,
+                'final': _list_of_packages
```

Comment:
(opinion) I think "runtime" is clearer than "final" (for example the explanatory comment in the code above this talks about "packages that are needed at runtime"). If the term "final" comes up frequently in the container community and has a well-understood meaning, then I think "final" is fine.

Reply:
The word "final" is somewhat used when talking of multi-stage builds (as in the "final image" etc. see [here](https://docs.docker.com/develop/develop-images/multistage-build/)). On the other hand the word "runtime" is used a lot when talking about container technologies with a complete different meaning (it's the mechanism by which a container image is run on the system), so I would avoid that particular term and maintain the current one.